src/PrefillGravityForms/Controllers/BaseController.php

```diff
@@ -353,6 +356,43 @@ protected function handleCurl(array $args): array
         }
     }
 
+    /**
+     * Extracts the burgerservicenummer (BSN) from the API response.
+     */
+    protected function extractBSN(array $response): string
+    {
+        return (string) ($response['burgerservicenummer'] ?? '');
+    }
+
+    /**
+     * Validates whether the necessary conditions are met before setting the transient.
+     *
+     * Ensures that:
+     * - A valid BSN (burgerservicenummer) is present in the response.
+     * - The transient key derived from that BSN matches the one generated from the current session.
+     */
+    protected function handleTransient(array $response, string $transientKey): void
+    {
+        $responseBSN = $this->extractBSN($response);
+
+        if ('' === $responseBSN) {
+            throw new Exception('No burgerservicenummer found in the response.', 404);
+        }
+
+        $transientKeyByResponse = CacheService::formatTransientKey($responseBSN);
+
+        // Ensure the transient keys generated from the BSN out of the response and current session match.
+        if ($transientKeyByResponse !== $transientKey) {
+            throw new Exception('Transient key mismatch.', 500);
+        }
+
+        try {
+            CacheService::setTransient($transientKey, $response);
+        } catch (Exception $e) {
+            $this->logError('Failed to set transient: ' . $e->getMessage(), $e->getCode());
```

Comment:
Hier ook weer de vraag: Is deze exception zo vaak voorkomend dat we de zaak willen redden als dat gebeurt?

Reply:
Is dat iets dat je verwacht, dat de cache faalt? Of is dat iets dat misschien kan voorkomen? En dat, als het voorkomt, dat je dat meteen wilt merken en oplossen?

Reply:
Klopt want zonder caching wordt er bij veelvuldig gebruik van de BRP een 504 gegooid. Wat tevens de reden is voor deze change. Was n.a.v. een support case.

Reply:
Nee klopt, we kunnen meer context meegeven aan de logger naar Teams maar op zich heb ik hier geen stack trace nodig. O.b.v. de exception message weet je zo waar je moet zijn. Is niet ideaal maar ik verwijs weer naar de php8 branch. Die kan prima loggen naar Sentry door een action waaraan de `$e` wordt meegeven.